tests/machinery/views.py

```diff
@@ -0,0 +1,81 @@
+
+import json
+import urlparse
+
+import pytest
+
+import requests_mock
+
+import caighdean
+
+from django.core.urlresolvers import reverse
+
+from pontoon.base.models import Entity, Locale, Translation
+
+
+@pytest.mark.django_db
+def test_view_mt_caighdean(client, entity0):
+    gd = Locale.objects.get(code='gd')
+    url = reverse('pontoon.machine_translation_caighdean')
+
+    response = client.get(url, dict(id=entity0.id))
+    assert json.loads(response.content) == {}
+
+    translation = Translation.objects.create(
+        entity=entity0, locale=gd, string='GD translation')
+    entity0.translation_set.add(translation)
+
+    translator = caighdean.Translator()
+
+    with requests_mock.mock() as m:
+        m.post(translator.service_url, text='[["source", "target"]]')
+        response = client.get(url, dict(id=entity0.id))
+    assert (
+        json.loads(response.content)
+        == {"translation": "target",
+            "original": translation.string})
+    assert (
+        urlparse.parse_qs(m.request_history[0].text)
+        == {u'teacs': [translation.string],
+            u'foinse': [gd.code]})
+
+
+@pytest.mark.django_db
+def test_view_mt_caighdean_bad(client, entity0):
+    gd = Locale.objects.get(code='gd')
+    url = reverse('pontoon.machine_translation_caighdean')
+
+    response = client.get(url)
+    assert response.status_code == 400
+    assert (
+        response.content
+        == 'Bad Request: "\'id\'"')
+
+    response = client.get(url, dict(id="DOESNOTEXIST"))
+    assert response.status_code == 400
+    assert (
+        response.content
+        == ("Bad Request: invalid literal for int() "
+            "with base 10: 'DOESNOTEXIST'"))
+
+    maxid = Entity.objects.values_list(
+        "id", flat=True).order_by("-id").first()
```

Comment:
This is a good example of why I dislike not putting closing brackets/parentheses on a new line. It is quite difficult to understand where the closing parenthesis of the first line is. To me, this is much more readable:

```python
maxid = Entity.objects.values_list(
    "id", flat=True
).order_by("-id").first()
```

I don't have to think to see where the function's arguments end, which is not the case with the way you wrote it.